Contact

EricChavarria opened the pull request:
<img width="256" alt="Screen Shot 2020-01-18 at 3 16 27 PM" src="https://user-images.githubusercontent.com/27781451/72669952-e0dc4180-3a05-11ea-8dd5-8de943099456.png">
<img width="399" alt="Screen Shot 2020-01-18 at 3 17 57 PM" src="https://user-images.githubusercontent.com/27781451/72669955-e89be600-3a05-11ea-8f24-cfade2ade89b.png">
<img width="474" alt="Screen Shot 2020-01-18 at 3 18 02 PM" src="https://user-images.githubusercontent.com/27781451/72669958-ed609a00-3a05-11ea-9957-07f82cebc835.png">
<img width="425" alt="Screen Shot 2020-01-18 at 3 18 23 PM" src="https://user-images.githubusercontent.com/27781451/72669962-f18cb780-3a05-11ea-9a96-8960bfc6d868.png">
So this is the contact page. I didn't do anything to the fonts because we don't exactly know what they are going to be yet. The photo is from the about page because we do not know what photo she want's here yet. The first photo is the mockup and the next two photos are the page I made. I couldn't fit them in one screenshot. The last photo is something I added. when you fill out all the fields and click send this message pops up on the screen.

Commit:
Made the contact page

Haon1919 commented on src/pages/Contact.tsx:
```diff
@@ -1,9 +1,42 @@
+            <img className={styles.contact_image} src={whitneyAbout} alt="Whitney Twigg's Photo"/>
+            <h1 className={styles.heading}>Contact Whitney</h1>
+            <p className={styles.information}>I'd love to hear from you! You can email me at twiggmusicstudio@gmail.com or through the form below. There's no such thing as a silly question.</p>
+            <form className={styles.contact_form} style={{"display":(contactFormSubmitted ? "none" : "flex")}} onSubmit={submitContactForm}>
```
Put these styles in a class and conditionally apply the class.

Haon1919 commented on src/pages/Contact.tsx:
```diff
@@ -1,9 +1,42 @@
+            <form className={styles.contact_form} style={{"display":(contactFormSubmitted ? "none" : "flex")}} onSubmit={submitContactForm}>
+                <div className={styles.name_field}>
+                    <label className={styles.contact_labels} htmlFor="name"><b>First and Last Name</b></label>
+                    <input id="name" name="name" type="text" required onChange={e => setName(e.target.value)}/>
```
Did you test this? This will not work. You need to have a type on the event object and I think it is e.currentTarget.value. Talk to me though if it does. It also needs a value={name}

Haon1919 commented on src/pages/Contact.tsx:
```diff
@@ -1,9 +1,42 @@
+                </div>
+                <div className={styles.email_field}>
+                    <label className={styles.contact_labels} htmlFor="email"><b>Email</b></label>
+                    <input id="email" name="email" type="text" required onChange={e => setEmail(e.target.value)}/>
```
Why do you have an ID on email if you have a unique classname on the parent element? Also add the type to the event and give it a value of email.

Haon1919 commented on src/pages/Contact.tsx:
```diff
@@ -1,9 +1,42 @@
+                </div>
+                <div className={styles.message_field}>
+                    <label className={styles.contact_labels} htmlFor="message"><b>Message</b></label>
+                    <textarea id="message" name="message" required onChange={e => setMessage(e.target.value)}/>
```
same as last onChange. You also need to give the text area a value of message.

Haon1919 commented on src/pages/Contact.tsx:
```diff
@@ -1,9 +1,42 @@
-
+            <img className={styles.contact_image} src={whitneyAbout} alt="Whitney Twigg's Photo"/>
+            <h1 className={styles.heading}>Contact Whitney</h1>
+            <p className={styles.information}>I'd love to hear from you! You can email me at twiggmusicstudio@gmail.com or through the form below. There's no such thing as a silly question.</p>
```
We will eventually have to make this text dynamic I believe.

Haon1919 reviewed (changes requested):
Look at the comments I left in the code and we can talk about them if you have any questions.

Commit:
Make conditional class

Commit:
Create Services Page

Commit:
get rid of id's

Haon1919 committed:
Merge branch 'develop' into contact

Commit:
Use the FormInput Component

Haon1919 commented on src/pages/Contact.tsx:
```diff
@@ -16,25 +22,62 @@ export const Contact : React.FC= () => {
+                >
+                <input/>
+                </FormInput>
+                <FormInput
```
Add email validation pattern. This can be found in the create account page to see how I did it.

Haon1919 committed:
Calendar html + render helper

Begin html and add a render function to construct the calendar with proper dates for the respective month.

Haon1919 commented on src/pages/Contact.tsx:
```diff
@@ -16,25 +22,62 @@ export const Contact : React.FC= () => {
+                >
+                <textarea/>
+                </FormInput>
                 <input className={styles.send} type="submit" value="Send"/>
```
Change to be the same color as the buttons on the create account page and add the disabled styles I have.

Haon1919 commented on src/styles/ContactTextArea.module.scss:
```diff
@@ -0,0 +1,33 @@
+.input_container{
```
This all could go inside of the Contact.module.scss file I think because you are able to pass the styles object into the FormInput component.

joverton2697 committed:
Landing Page

Landing page is mostly finished, final images and text need to be decided. Header has blank space underneath

joverton2697 committed:
Implementing feedback

Commit:
Make changes to Contact Page

Commit:
Make styling change to services

Commit:
Change content

Commit:
Get started on Blog Page

Commit:
Finish up Blog Component

Commit:
get Started on Forgot Password Page

Commit:
Finish Forgot Password Page

Haon1919 committed:
Got the repository battle ready

Haon1919 committed:
Got the repository battle ready

Finish calendar, add schedule lesson and lesson page, fix header z-index issue, add fake login functionality.

Haon1919 committed:
Merge remote-tracking branch 'origin/Calendar' into Calendar

Haon1919 committed:
Merge remote-tracking branch 'origin/blog' into PresentationBranch

Haon1919 committed:
Merge branch 'forgot-password' into PresentationBranch

Haon1919 committed:
Merge remote-tracking branch 'origin/Services' into PresentationBranch

Haon1919 committed:
Merge remote-tracking branch 'origin/contact' into PresentationBranch

Haon1919 committed:
Merge remote-tracking branch 'origin/Landing' into PresentationBranch

Haon1919 committed:
Fix Login link in header

Fix login link in header by prefacing it with a slash

Commit:
Fix typo and mini bug

Haon1919 committed:
General tab remains collapsed when user logs out

Fix bug where if the user logs out and the Students tab is open, the general tab remains collapsed.

Haon1919 committed:
Merge remote-tracking branch 'origin/PresentationBranch' into PresentationBranch

Haon1919 committed:
Nonrequired form blocking submit if fails validation

If a non required field fails validation, make sure that when they remove the text it removes the failing field from the list of failing fields.

Haon1919 committed:
Minor form failing error persists for non minors

Fix error where when the birth date of the user made them a minor,and the user entered an invalid email for the guardian email then switched the birth date to make them an adult and filled in all of the other forms correctly, the  invalid guardian email still disallowed them from submitting even though they guardian email is not an option for adults.

Haon1919 committed:
Merge branch 'PresentationBranch' into contact
